plugins/crypto/openssl/securitypolicy_aes128sha256rsaoaep.c

```diff
@@ -121,6 +121,12 @@ updateCertificateAndPrivateKey_sp_aes128sha256rsaoaep(UA_SecurityPolicy *securit
     Policy_Context_Aes128Sha256RsaOaep *pc =
         (Policy_Context_Aes128Sha256RsaOaep *)securityPolicy->policyContext;
 
+    UA_Boolean isLocalKey = false;
+    if(newPrivateKey.length <= 0) {
+        if(UA_CertificateUtils_comparePublicKeys(&newCertificate, &securityPolicy->localCertificate) == 0)
+            isLocalKey = true;
+    }
```

Comment:
This somehow looks like a workaround because the information whether to use the current private key or the one from the latest CSR is not passed to the callback. But this information has already been gathered in UA_Server_updateCertificate().